example-buzz/src/lib.rs

```diff
@@ -18,62 +18,77 @@ use std::sync::Arc;
 use zenoh_flow::{bail, prelude::*};
 use zenoh_flow_example_types::{ZFString, ZFUsize};
 
-struct BuzzOperator;
+struct BuzzOperator {
+    word: String,
+    input_value: Input,
+    input_fizz: Input,
+    output_buzz: Output,
+}
 
 static LINK_ID_INPUT_INT: &str = "Int";
 static LINK_ID_INPUT_STR: &str = "Str";
 static LINK_ID_OUTPUT_STR: &str = "Str";
 
 #[async_trait]
-impl Operator for BuzzOperator {
-    async fn setup(
+impl Node for BuzzOperator {
+    async fn iteration(&self) -> Result<()> {
+        let value = match self.input_value.recv_async().await.unwrap() {
+            Message::Data(mut msg) => msg.get_inner_data().try_get::<ZFUsize>()?.clone(),
+            _ => bail!(ErrorKind::InvalidData, "No data"),
+        };
+
+        let fizz = match self.input_fizz.recv_async().await.unwrap() {
+            Message::Data(mut msg) => msg.get_inner_data().try_get::<ZFString>()?.clone(),
+            _ => bail!(ErrorKind::InvalidData, "No data"),
+        };
+
+        let mut buzz = fizz.clone();
+        if value.0 % 3 == 0 {
+            buzz.0.push_str(&self.word);
+        }
+
+        self.output_buzz.send_async(Data::from(buzz), None).await?;
```

Comment:
I think it would be better to use `select!` here.
Something like:

```rust
select! {
                    msg = self.input_value.recv_async().fuse() => {
                        if let Ok(Message::Data(mut msg)) = msg {
                            let inner_data = msg.get_inner_data().try_get::<ZFUsize>()?;
                            // do something, eg. store the value in self
                        }
                    },
                    msg = self.input_fizz.recv_async().fuse() => {
                        if let Ok(Message::Data(mut msg)) = msg {
                            let inner_data = msg.get_inner_data().try_get::<ZFString>()?;
                            // do something
                            self.output_buzz.send_async(Data::from(buzz), None).await?;
                        }
                    },       
             }
```